### ISSUE TITLE
NPE when table is bucketed on non-existent column

### ISSUE THREAD

**Issue by martint** [prestodb/presto#6553]
Given this table (which is, allegedly, bucketed on a non-existent column)

```
CREATE TABLE foo (a bigint) 
WITH (bucketed_by = array['b'], bucket_count=1000)
```

Querying the table fails with:

```
java.lang.NullPointerException
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.facebook.presto.hive.HiveMetadata.lambda$getTableLayout$6(HiveMetadata.java:1037)
	at java.util.Optional.map(Optional.java:215)
	at com.facebook.presto.hive.HiveMetadata.getTableLayout(HiveMetadata.java:1031)
	at com.facebook.presto.hive.HiveMetadata.getTableLayouts(HiveMetadata.java:999)
	at com.facebook.presto.hive.PrismMetadata.getTableLayouts(PrismMetadata.java:386)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorMetadata.getTableLayouts(ClassLoaderSafeConnectorMetadata.java:69)
	at com.facebook.presto.metadata.MetadataManager.getLayouts(MetadataManager.java:298)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planTableScan(AddExchanges.java:583)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitTableScan(AddExchanges.java:535)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitTableScan(AddExchanges.java:197)
	at com.facebook.presto.sql.planner.plan.TableScanNode.accept(TableScanNode.java:135)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.planChild(AddExchanges.java:1169)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitOutput(AddExchanges.java:244)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges$Rewriter.visitOutput(AddExchanges.java:197)
	at com.facebook.presto.sql.planner.plan.OutputNode.accept(OutputNode.java:81)
	at com.facebook.presto.sql.planner.optimizations.AddExchanges.optimize(AddExchanges.java:147)
	at com.facebook.presto.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:118)
	at com.facebook.presto.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:109)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:288)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:267)
	at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:225)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

This is related to https://github.com/prestodb/presto/issues/6552

**Comment by martint** [comment]
cc @haozhun

**Comment by haozhun** [comment]
Fixed in #6582
